### PR TITLE
add usecases.md and fix some typos

### PR DESCRIPTION
**Pull request by duanjiong**
Signed-off-by: Duan Jiong <djduanjiong@gmail.com>

### CODE REVIEW

File: doc/usecases.md [hunk 1]
```diff
@@ -0,0 +1,57 @@
+#layer2
+
+Usage can refer to this [doc](layer2.md)
+
+## precondition
+
+The cluster should have at least two nodes, nodeA and nodeB.
+
+## test endpoint update
+1. create service test-svc which have one endpoint on nodeA.
```
Comment (zheng1) [inline doc/usecases.md:10]: Please add some yaml here

File: doc/usecases.md [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+2. can successfully access to test-svc
```
Comment (zheng1) [inline doc/usecases.md:11]: And add some command about `kubectl exec xxxxxxx`

File: doc/usecases.md (hunk 1 above)
Comment (zheng1) [reply doc/usecases.md:10]: Just like this document: https://github.com/ahmetb/kubernetes-network-policy-recipes/blob/master/01-deny-all-traffic-to-an-application.md#example

File: doc/usecases.md [hunk 3]
```diff
@@ -0,0 +1,188 @@
+#Layer2
+
+Usage can refer to this [doc](layer2.md)
+
+## Precondition
```
Comment (FeynmanZhou) [inline doc/usecases.md:5]: Precondition  => Prerequiste, `Precondition` seems not that common-used in technical doc.

File: doc/eip_config.md [hunk 4]
```diff
@@ -0,0 +1,21 @@
+#Eip Configuration
```
Comment (FeynmanZhou) [inline doc/eip_config.md:1]: EIP

File: doc/eip_config.md [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+
+Porter only support ipv4 address now. Let's take this as an example
```
Comment (FeynmanZhou) [inline doc/eip_config.md:3]: supports

File: doc/eip_config.md [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+apiVersion: network.kubesphere.io/v1alpha1
+kind: Eip
+metadata:
+    name: eip-sample-pool
+spec:
+    address: 192.168.0.0/24
+    protocol: layer2
+```
+
+1. protocol should be layer2 or bgp.
```
Comment (FeynmanZhou) [inline doc/eip_config.md:15]: The protocol should be layer 2 or BGP.

File: doc/eip_config.md [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+2. address now support 3 kinds of address syntax
```
Comment (FeynmanZhou) [inline doc/eip_config.md:16]: The address supports three kinds of syntax.

File: doc/usecases.md [hunk 8]
```diff
@@ -0,0 +1,188 @@
...
+
+Usage can refer to this [doc](layer2.md)
+
+## Prerequisite
+
+The cluster should have at least two nodes, nodeA and nodeB.
+
+## Test endpoint update
+
+- create service test-svc which have one endpoint on nodeA.
```
Comment (FeynmanZhou) [inline doc/usecases.md:11]: Create a Service `test-svc` which has one endpoint on nodeA.

File: doc/usecases.md [hunk 9]
```diff
@@ -0,0 +1,188 @@
...
+        app:  mylbapp
+    type:  LoadBalancer 
+    ports:
+      - name:  http
+        port:  80
+        targetPort:  80
+```
+
+- exec 'wget ${eip}' can successfully access to test-svc
+- exec 'ip neigh  show to ${eip} dev ${nic}', we can find lladdr is nodeA's Mac
```
Comment (FeynmanZhou) [inline doc/usecases.md:46]: what does `lladdr ` mean?

File: doc/layer2.md [hunk 10]
```diff
@@ -1,4 +1,5 @@
 #Preparation
+
 Normally, Nic will answer arp request which ip not reside on it. In order to let porter-manager
```
Comment (FeynmanZhou) [inline doc/layer2.md:3]: arp => ARP.

File: doc/usecases.md [hunk 11]
```diff
@@ -0,0 +1,188 @@
+#Layer2
+
+Usage can refer to this [doc](layer2.md)
```
Comment (FeynmanZhou) [inline doc/usecases.md:3]: [Layer 2 guide](layer2.md) demonstrates how to install Porter on Kubernetes with Layer 2 protocol.

File: doc/usecases.md [hunk 12]
```diff
@@ -0,0 +1,188 @@
...
+    imagePullPolicy: IfNotPresent
+  nodeSelector:
+    kubernetes.io/hostname: nodeB
+```
+- exec 'wget ${eip}' can successfully access to test-svc
+- exec 'ip neigh  show to ${eip} dev ${nic}', we can find lladdr is nodeB's Mac
+
+# BGP
+
+Usage can refer to this [doc](simulate_with_bird.md)
```
Comment (FeynmanZhou) [inline doc/usecases.md:68]: [BGP guide](simulate_with_bird.md) demonstrates how to install Porter and use BGP on Kubernetes.

File: doc/usecases.md [hunk 13]
```diff
@@ -0,0 +1,188 @@
...
+
+## Prerequisite
+
+The cluster should have at least two nodes, nodeA and nodeB.
+
+## Test BGP PortForward
+- config BgpConf with port 17900
```
Comment (FeynmanZhou) [inline doc/usecases.md:75]: config => Configure

File: doc/usecases.md [hunk 14]
```diff
@@ -0,0 +1,188 @@
...
+kind: BgpConf
+metadata:
+  name: bgpconf-sample
+spec:
+  # Add fields here
+  as : 65000
+  routerID : 192.168.0.2
+  port: 17900
+```
+- config BgpPeer with usingPortForward
```
Comment (FeynmanZhou) [inline doc/usecases.md:87]: config => Configure

File: doc/usecases.md [hunk 15]
```diff
@@ -0,0 +1,188 @@
...
+spec:
+  # Add fields here
+  usingPortForward: true
+  config:
+    peerAs : 65001
+    neighborAddress: 192.168.0.6
+  addAaths:
+    sendMax: 10
+```
+- config bird  neighbour  with port 179
```
Comment (FeynmanZhou) [inline doc/usecases.md:102]: config => Configure

File: doc/usecases.md [hunk 16]
```diff
@@ -0,0 +1,188 @@
...
+- exec 'birdctl show protocol', the nighbour's state should be up
```
Comment (FeynmanZhou) [inline doc/usecases.md:103]: Replace all `exec` to `Execute`

File: doc/usecases.md [hunk 17]
```diff
@@ -0,0 +1,188 @@
...
+
+## Test BGP passive mode
+- config BgpConf with port 17900
```
Comment (FeynmanZhou) [inline doc/usecases.md:106]: Replace all `config` to `Configure` in the sentences.

File: doc/usecases.md [hunk 18]
```diff
@@ -0,0 +1,188 @@
...
+spec:
+  config:
+    peerAs : 65001
+    neighborAddress: 192.168.0.6
+  addAaths:
+    sendMax: 10
+  transport:
+    passiveMode: true
+```
+- config bird neighbour with port 17900
```
Comment (FeynmanZhou) [inline doc/usecases.md:122]: Same as above

File: doc/usecases.md [hunk 19]
```diff
@@ -0,0 +1,188 @@
...
+    neighborAddress: 192.168.0.6
+  addAaths:
+    sendMax: 10
+```
+- config bird  neighbour  with port 179
+- exec 'birdctl show protocol', the nighbour's state should be up
+
+## Test BGP passive mode
+- config BgpConf with port 17900
+- config BgpPeer with passiveMode true
```
Comment (FeynmanZhou) [inline doc/usecases.md:107]: Set `passiveMode ` to `true` in BgpPeer configuration.

File: doc/usecases.md [hunk 20]
```diff
@@ -0,0 +1,188 @@
...
+  addAaths:
+    sendMax: 10
+  transport:
+    passiveMode: true
+```
+- config bird neighbour with port 17900
+- exec 'birdctl show protocol', the nighbour's state should be up
+
+## Test endpoint update
+- create service test-svc which have two endpoints on nodeA and nodeB.
```
Comment (FeynmanZhou) [inline doc/usecases.md:126]: - Create a Service `test-svc` which has two endpoints on nodeA and nodeB respectively.

File: doc/usecases.md [hunk 21]
```diff
@@ -0,0 +1,188 @@
...
+spec:
+    selector:
+        app:  mylbapp
+    type:  LoadBalancer 
+    ports:
+      - name:  http
+        port:  80
+        targetPort:  80
+```
+- exec 'wget ${eip}' can successfully access to test-svc
```
Comment (FeynmanZhou) [inline doc/usecases.md:173]: Rarely see this written method. Why not put the command in a single line? 

Example:

Execute the following command, verify if you can successfully access the Service `test-svc`:

```
wget ${eip}
```

Comment (duanjiong) [reply doc/usecases.md:173]: This document is basically a test document ,  every line has a command and it's repeated,  so i think it is not a good way to write it like you comment above.

**Pull request merged**